File: src-tauri/src/models/note.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct NoteListItem {
    pub id: String,
    pub title: String,
    pub path: String,
    pub modified: String,
    pub tags: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RawNote {
    pub path: String,
    pub content: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SearchResult {
    pub id: String,
    pub path: String,
    pub title: String,
    pub snippet: String,
    pub tags: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct BacklinkItem {
    pub id: String,
    pub path: String,
    pub title: String,
    pub modified: String,
}

File: src/types/note.ts
export interface Frontmatter {
  id: string;
  title: string;
  created: string;
  modified: string;
  tags: string[];
  links: string[];
  [key: string]: unknown;
}

export interface Note {
  path: string;
  frontmatter: Frontmatter;
  body: string;
}

export interface NoteListItem {
  id: string;
  title: string;
  path: string;
  modified: string;
  tags: string[];
}

export interface SearchResult {
  id: string;
  path: string;
  title: string;
  snippet: string;
  tags: string[];
}
